designs/full-cluster-lifecycle-api-cloudstack.md

```diff
@@ -0,0 +1,81 @@
+# Cluster Lifecycle API CloudStack
+
+## Introduction
+
+### Problem
+
+Currently the only way to manage EKS Anywhere CloudStack cluster is by using CLI. We need to support native K8s experience on CloudStack cluster lifecycle.
+
+### Goals and Objectives
+
+* Create/Upgrade/Delete CloudStack workload clusters using: kubectl apply -f cluster.yaml
+* Create/Update/Upgrade/Delete CloudStack workload clusters using GitOps/Terraform
+
+#### Not in Scope
+
+* Create/Upgrade/Delete self-managed clusters with API
+
+#### Limitation
+
+* We currently only supports k8s version < 1.24 on CloudStack
```

Comment:
I don't think this is true anymore
Also, is this really a limitation when designing this feature? It seems like design is decoupled from the kubernetes version and works transparently to it

Reply:
https://github.com/aws/eks-anywhere/blob/7d229f96bd177e317e943e3891e75d21ad54a937/pkg/providers/cloudstack/cloudstack.go#L1340

Reply:
I'll delegate that decision to @vignesh-goutham but afaik, 1.24 works with our latest release.

Reply:
Our current capc version is 0.4.8-rc1, which doesn’t support k8s version > 1.24, that's why we have such validation.
But capc latest version 0.4.9-rc4 just support k8s 1.24 ([pr](https://github.com/kubernetes-sigs/cluster-api-provider-cloudstack/pull/189))
@rahulbabu95 is currently testing 1.25 and above

> from 1.24 k8s started using nft instead of iptables for its networking configuration. and guillermo had to change kube-proxy to support that. so now technically we should be able to create 125 cluster in CS but yeah running into some issues which me and vgg are looking into